app/Keycloak/Client/KeycloakApiClient.php

```diff
@@ -111,13 +112,13 @@ public function deleteScopes(Client $client): void
     /**
      * @throws KeyCloakApiFailed
      */
-    public function fetchClient(Realm $realm, Integration $integration): Client
+    public function fetchClient(Realm $realm, Integration $integration, UuidInterface $clientId): Client
     {
         try {
             $response = $this->client->sendWithBearer(
                 new Request(
                     'GET',
-                    'admin/realms/' . $realm->internalName . '/clients?' . http_build_query(['clientId' => $integration->id->toString()])
+                    'admin/realms/' . $realm->internalName . '/clients?' . http_build_query(['clientId' => $clientId->toString()])
```

Comment:
According the API docs all calls can be done with the id field instead of the clientId